homeassistant/components/litterrobot/entity.py

```diff
@@ -23,17 +23,20 @@
 from .hub import LitterRobotHub
 
 _P = ParamSpec("_P")
+_RobotT = TypeVar("_RobotT", bound=Robot)
 _LOGGER = logging.getLogger(__name__)
 
 REFRESH_WAIT_TIME_SECONDS = 8
 
 
-class LitterRobotEntity(CoordinatorEntity[DataUpdateCoordinator[bool]]):
+class LitterRobotEntity(
+    CoordinatorEntity[DataUpdateCoordinator[bool]], Generic[_RobotT]
+):
```

Comment:
Thanks, this was the primary piece I was missing when trying it on my own and couldn't figure it out for the life of me and thought maybe I was doing it completely wrong 🤦🏻

Reply:
I'll update the other 3 commits to match once this one gets merged.

Reply:
As I do not own any litterrobot, I have not been able to run any tests and I have relied on pytest to pickup mistakes.

Reply:
All is in working order. Thanks again!